### PR TITLE
isort, slight docstring formatting

### PR DESCRIPTION
**Pull request by tomashozman**
Hello again :slightly_smiling_face: I've just gone for the auto-formatting route this time, except for one change.

I've run isort & Black over all of the Python version of Oso (in that order,) removed the newlines under a few function docstrings that had them as per PEP 257. Black, as far as I know, doesn't have any guide on these, so it shouldn't interfere with anything.

Now, for the manual change - I've removed the newlines between `:param:`s in `oso/oso.py`, as they weren't anywhere else - for consistency's sake and all that. Also extra newlines at the end of 2 docstrings in that same file.

I do hope nothing breaks this time.

### CODE REVIEW

**Review by samscott89 — DISMISSED** [review]
Thanks @tomashozman! Thanks for making us aware of `isort`. I didn't know it existing, and seems like a good thing to add.

This looks good, but we should make it part of the development process so it stays up to date. We also run a bunch of formatting checks through our CI process.

The places we would want to add this:

- [ ] `make fmt` in oso/: https://github.com/osohq/oso/blob/main/languages/python/oso/Makefile#L18
- [ ] `make fmt` in flask-oso/ https://github.com/osohq/oso/blob/main/languages/python/flask-oso/Makefile#L16
- [ ] `make fmt` in sqlalchemy-oso/ https://github.com/osohq/oso/blob/main/languages/python/sqlalchemy-oso/Makefile#L22
- [ ] `make fmt` in django-oso https://github.com/osohq/oso/blob/main/languages/python/django-oso/Makefile#L20
- [ ] Add to the fmt requirements: https://github.com/osohq/oso/blob/main/languages/python/requirements-fmt.txt
- [ ] Add to `test.yml` https://github.com/osohq/oso/blob/main/.github/workflows/test.yml#L53 This would need to `pip install isort` and run `isort languages/python -c`

Happy to help with this if you would like!

**Comment by tomashozman** [comment]
I'll add it in a little bit. :slightly_smiling_face: One thing to note though, `isort`'s default formatting collides with the Black code style.
Instead of 
```
from [   ] import (
    thing1,
    thing2,
    thing3,
    thing4
)
```
, it does
```
from [     ] import (thing1, thing2, thing3,
                            thing4)
```
. It's possible to fix that by putting it before other auto-formatting tools, but it still feels a bit hacky. Perhaps there's a setting somewhere to change that behavior?

**Comment by samscott89** [comment]
> 
> . It's possible to fix that by putting it before other auto-formatting tools, but it still feels a bit hacky. Perhaps there's a setting somewhere to change that behavior?

Oh, good call!

Seems like there's configuration for it: https://pycqa.github.io/isort/docs/configuration/black_compatibility.html#using-a-config-file-such-as-isortcfg

We have a few setup.cfg files around that we could put that option?

**Comment by tomashozman** [comment]
Ah, that's easier than I thought. I'll add that as well then.

**Comment by tomashozman** [comment]
Oh, where did the `.github` one go? I could've sworn the commit had that. Let's see..

File: languages/python/oso/setup.cfg [hunk 1]
```diff
@@ -18,3 +18,6 @@ ignore_missing_imports = True
 max-line-length = 127
 extend-exclude = build/
 per-file-ignores = __init__.py: F401
+
+[isort]
+profile=black
```
Comment (samscott89) [inline languages/python/oso/setup.cfg:23]: Will every package need this?

**Comment by tomashozman** [comment]
Ah, you can tell the `isort` check works now. I guess it's time to format everything with it?

**Comment by tomashozman** [comment]
I see, it doesn't recognize the code style.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -47,10 +47,14 @@ jobs:
...
-      - name: Install Python formatter
+      - name: Install Python Black code style formatter
         run: pip install black
-      - name: Check Python formatting
+      - name: Install Python isort formatter
+        run: pip install isort
+      - name: Check Black formatting
         run: black --check languages/python
+      - name: Check isort formatting
+        run: isort languages/python -c
```
Comment (samscott89) [inline .github/workflows/test.yml:57]:
```suggestion
        run: isort languages/python -c  --profile black
```
Perhaps?

Comment (tomashozman) [reply .github/workflows/test.yml:57]: That's indeed probably the better option :smile:

**Review by dhatch — APPROVED** [review]

**Comment by dhatch** [comment]
I'm finding that running `isort` locally doesn't seem to be consistent with the tests. I haven't had a chance to figure out why yet.

**Review by dhatch — CHANGES_REQUESTED** [review]
```
(venv) ➜  oso git:(f6e686a8) ✗ make -C languages/python fmt
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C oso fmt
isort .
Skipped 3 files
black .
All done! ✨ 🍰 ✨
31 files left unchanged.
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C django-oso fmt
isort .
Skipped 1 files
black .
All done! ✨ 🍰 ✨
20 files left unchanged.
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C flask-oso fmt
isort .
Skipped 1 files
black .
All done! ✨ 🍰 ✨
6 files left unchanged.
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C sqlalchemy-oso fmt
isort .
Skipped 2 files
black .
All done! ✨ 🍰 ✨
21 files left unchanged.
(venv) ➜  oso git:(f6e686a8) ✗ isort languages/python -c --profile black
ERROR: /Users/dhatch/code/oso/languages/python/sqlalchemy-oso/sqlalchemy_oso/auth.py Imports are incorrectly sorted and/or formatted.
ERROR: /Users/dhatch/code/oso/languages/python/sqlalchemy-oso/sqlalchemy_oso/roles.py Imports are incorrectly sorted and/or formatted.
ERROR: /Users/dhatch/code/oso/languages/python/sqlalchemy-oso/sqlalchemy_oso/session.py Imports are incorrectly sorted and/or formatted.
ERROR: /Users/dhatch/code/oso/languages/python/sqlalchemy-oso/sqlalchemy_oso/partial.py Imports are incorrectly sorted and/or formatted.
ERROR: /Users/dhatch/code/oso/languages/python/sqlalchemy-oso/tests/conftest.py Imports are incorrectly sorted and/or formatted.
ERROR: /Users/dhatch/code/oso/languages/python/sqlalchemy-oso/tests/test_sqlalchemy.py Imports are incorrectly sorted and/or formatted.
ERROR: /Users/dhatch/code/oso/languages/python/sqlalchemy-oso/tests/test_roles_old.py Imports are incorrectly sorted and/or formatted.
ERROR: /Users/dhatch/code/oso/languages/python/sqlalchemy-oso/tests/test_post_relationship.py Imports are incorrectly sorted and/or formatted.
ERROR: /Users/dhatch/code/oso/languages/python/sqlalchemy-oso/tests/test_roles.py Imports are incorrectly sorted and/or formatted.
ERROR: /Users/dhatch/code/oso/languages/python/sqlalchemy-oso/tests/test_partial_parity.py Imports are incorrectly sorted and/or formatted.
ERROR: /Users/dhatch/code/oso/languages/python/sqlalchemy-oso/tests/test_flask.py Imports are incorrectly sorted and/or formatted.
ERROR: /Users/dhatch/code/oso/languages/python/sqlalchemy-oso/tests/test_preprocess.py Imports are incorrectly sorted and/or formatted.
ERROR: /Users/dhatch/code/oso/languages/python/flask-oso/flask_oso/context.py Imports are incorrectly sorted and/or formatted.
ERROR: /Users/dhatch/code/oso/languages/python/flask-oso/flask_oso/flask_oso.py Imports are incorrectly sorted and/or formatted.
ERROR: /Users/dhatch/code/oso/languages/python/flask-oso/tests/test_flask_oso.py Imports are incorrectly sorted and/or formatted.
ERROR: /Users/dhatch/code/oso/languages/python/django-oso/tests/test_django_oso.py Imports are incorrectly sorted and/or formatted.
ERROR: /Users/dhatch/code/oso/languages/python/django-oso/tests/test_post_relationship.py Imports are incorrectly sorted and/or formatted.
ERROR: /Users/dhatch/code/oso/languages/python/django-oso/tests/test_partial_parity.py Imports are incorrectly sorted and/or formatted.
ERROR: /Users/dhatch/code/oso/languages/python/django-oso/tests/test_urls.py Imports are incorrectly sorted and/or formatted.
ERROR: /Users/dhatch/code/oso/languages/python/django-oso/tests/test_app/models.py Imports are incorrectly sorted and/or formatted.
ERROR: /Users/dhatch/code/oso/languages/python/django-oso/tests/test_app2/models.py Imports are incorrectly sorted and/or formatted.
ERROR: /Users/dhatch/code/oso/languages/python/django-oso/django_oso/models.py Imports are incorrectly sorted and/or formatted.
ERROR: /Users/dhatch/code/oso/languages/python/django-oso/django_oso/oso.py Imports are incorrectly sorted and/or formatted.
ERROR: /Users/dhatch/code/oso/languages/python/django-oso/django_oso/partial.py Imports are incorrectly sorted and/or formatted.
ERROR: /Users/dhatch/code/oso/languages/python/django-oso/django_oso/middleware.py Imports are incorrectly sorted and/or formatted.
ERROR: /Users/dhatch/code/oso/languages/python/oso/tests/conftest.py Imports are incorrectly sorted and/or formatted.
ERROR: /Users/dhatch/code/oso/languages/python/oso/tests/test_polar.py Imports are incorrectly sorted and/or formatted.
ERROR: /Users/dhatch/code/oso/languages/python/oso/tests/test_polar_roles.py Imports are incorrectly sorted and/or formatted.
ERROR: /Users/dhatch/code/oso/languages/python/oso/tests/test_repr.py Imports are incorrectly sorted and/or formatted.
ERROR: /Users/dhatch/code/oso/languages/python/oso/tests/test_oso.py Imports are incorrectly sorted and/or formatted.
ERROR: /Users/dhatch/code/oso/languages/python/oso/tests/parity/test_polar.py Imports are incorrectly sorted and/or formatted.
ERROR: /Users/dhatch/code/oso/languages/python/oso/tests/parity/test_api.py Imports are incorrectly sorted and/or formatted.
Skipped 9 files
```

`isort` not changing any files on `make fmt` doesn't seem to be consistent with the check results showing changes are required.

**Comment by tomashozman** [comment]
That certainly is weird. I'm not sure why that's happening either.

**Comment by dhatch** [comment]
@tomashozman thank you for putting together this PR.

I'm going to close it for now because I'm not sure why `isort` is displaying different behavior on CI & locally. If you're able to reproduce this behavior locally and have a hypothesis for why it's occurring let me know and I can reopen the PR.

**Pull request closed without merging**